packages/iov-tendermint-rpc/src/client.spec.ts

```diff
@@ -208,197 +399,13 @@ describe("Client", () => {
-      pendingWithoutTendermint();
-
-      (async () => {
-        const events: responses.TxEvent[] = [];
-        const client = await Client.connect("ws://" + tendermintUrl);
-        const stream = client.subscribeTx();
-        expect(stream).toBeTruthy();
-        const subscription = stream.subscribe({
-          next: event => {
-            expect(event.height).toBeGreaterThan(0);
-            expect(event.index).toEqual(0);
-            expect(event.result).toBeTruthy();
-            expect(event.tx.length).toBeGreaterThan(10);
-
-            events.push(event);
-
-            if (events.length === 2) {
-              subscription.unsubscribe();
-              expect(events.length).toEqual(2);
-              expect(events[1].height).toEqual(events[0].height + 1);
-              expect(events[1].result.tags).not.toEqual(events[0].result.tags);
-              done();
-            }
-          },
-          error: fail,
-          complete: () => fail("Stream must not close just because we don't listen anymore"),
-        });
-
-        const transaction1 = buildKvTx(randomId(), randomId());
-        const transaction2 = buildKvTx(randomId(), randomId());
-
-        await client.broadcastTxCommit({ tx: transaction1 });
-        await client.broadcastTxCommit({ tx: transaction2 });
-      })().catch(fail);
-    });
-
-    it("can subscribe to transaction events filtered by creator", done => {
-      pendingWithoutTendermint();
-
-      (async () => {
-        const events: responses.TxEvent[] = [];
-        const client = await Client.connect("ws://" + tendermintUrl);
-        const tags: ReadonlyArray<QueryTag> = [{ key: "app.creator", value: "jae" }];
-        const stream = client.subscribeTx(tags);
-        expect(stream).toBeTruthy();
-        const subscription = stream.subscribe({
-          next: event => {
-            expect(event.height).toBeGreaterThan(0);
-            expect(event.index).toEqual(0);
-            expect(event.result).toBeTruthy();
-            expect(event.tx.length).toBeGreaterThan(10);
-
-            events.push(event);
-
-            if (events.length === 2) {
-              subscription.unsubscribe();
-              expect(events.length).toEqual(2);
-              expect(events[1].height).toEqual(events[0].height + 1);
-              expect(events[1].result.tags).not.toEqual(events[0].result.tags);
-              done();
-            }
-          },
-          error: fail,
-          complete: () => fail("Stream must not close just because we don't listen anymore"),
-        });
-
-        const transaction1 = buildKvTx(randomId(), randomId());
-        const transaction2 = buildKvTx(randomId(), randomId());
-
-        await client.broadcastTxCommit({ tx: transaction1 });
-        await client.broadcastTxCommit({ tx: transaction2 });
-      })().catch(fail);
-    });
+    const factory = () => new WebsocketClient(tendermintUrl, onError);
+    websocketTestSuite(factory, v0_25);
```

Comment:
This adds some strange nesting that is hard to follow. We can keep it flat as before:

```ts
const factory = () => new WebsocketClient(tendermintUrl, onError);
kvTestSuite(factory, v0_25);
websocketTestSuite(factory, v0_25);
```